otter/assign/solutions.py

```diff
@@ -208,21 +208,46 @@ def strip_solutions_and_output(nb):
     """
     nb = copy.deepcopy(nb)
 
-    md_solutions = []
+    del_md_solutions = []
     lang = get_notebook_language(nb)
     for i, cell in enumerate(nb['cells']):
         if has_tag(cell, SOLUTION_CELL_TAG):
             if is_cell_type(cell, "code"):
                 cell['source'] = '\n'.join(replace_solutions(get_source(cell), lang))
             elif is_cell_type(cell, "markdown"):
-                md_solutions.append(i)
+                if has_tag(cell, SOLUTION_INCLUDE_MD_TAG):
+                    remove_tag(cell, SOLUTION_INCLUDE_MD_TAG)
+                else:
+                    del_md_solutions.append(i)
             nb['cells'][i] = remove_tag(cell, SOLUTION_CELL_TAG)
 
-    md_solutions.reverse()
-    for i in md_solutions:
+    del_md_solutions.reverse()
+    for i in del_md_solutions:
         del nb['cells'][i]
 
     # remove output from student version
     remove_output(nb)
 
     return nb
+
+SOLUTION_INCLUDE_MD_TAG = "otter_assign_solution_include_md"
+INCLUDE_MD_REGEX = re.compile(r"<!---?\s*\n*\s*INCLUDESTUDENT\s*\n*\s*-->\s*\n")
```

Comment:
What is the purpose of requiring instructors to include `<!-- INCLUDESTUDENT -->` in the text of the markdown cell? Why not just have instructors add a tag to the cell like `otter_include` to the cell and just not remove the cell in the function above if that tag is present? That's the way other similar features are handled, and I'd prefer to use the same type of structure for this feature.

Reply:
When you say add tags you mean like View -> Cell Toolbar ->  Tags and then manually add the tags?

Reply:
Yes